### PR TITLE
[Java] DSM: Fix kafka test flakiness

### PR DESCRIPTION
**Pull request by tabgok**
## Motivation

Fix for flaky DSM tests with the Java app.

## Changes

Adds `thread.join()` calls to all the calls to produce/consume.  Also updates method signatures to allow this to happen.

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] [Relevant label](https://github.com/DataDog/system-tests/blob/main/docs/CI/labels.md) (`run-parametric-scenario`, `run-profiling-scenario`...) are presents
* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

### CODE REVIEW

File: utils/build/docker/java/spring-boot/src/main/java/com/datadoghq/system_tests/springboot/App.java [hunk 1]
```diff
@@ -463,14 +465,18 @@ String publishToKafka(
         if ("kafka".equals(integration)) {
             KafkaConnector kafka = new KafkaConnector(queue);
             try {
-                kafka.startProducingMessage("hello world!");
+                Thread produceThread = kafka.startProducingMessage("hello world!");
+                produceThread.join(this.PRODUCE_CONSUME_THREAD_TIMEOUT);
+                produceThread.join(this.PRODUCE_CONSUME_THREAD_TIMEOUT);
```
Comment (vandonr) [inline utils/build/docker/java/spring-boot/src/main/java/com/datadoghq/system_tests/springboot/App.java:470]: why 2 `join` here and on the next 3 ?

**Review by vandonr — COMMENTED** [review]
LGTM except for the duplicated lines

**Comment by cbeauchesne** [comment]
You can rebase your branch to fix failures on the CI

File: utils/build/docker/java/spring-boot/src/main/java/com/datadoghq/system_tests/springboot/App.java (hunk 1 above)
Comment (tabgok) [reply utils/build/docker/java/spring-boot/src/main/java/com/datadoghq/system_tests/springboot/App.java:470]: fixed - find+replace error

**Review by vandonr — APPROVED** [review]
good to merge when all tests pass

**Comment by robertomonteromiguel** [comment]
There are test failure related with dsm. I move this PR as draft

**Review by robertomonteromiguel — APPROVED** [review]

**Pull request merged**